src/app/components/apps-view/apps-view.component.ts

```diff
@@ -111,19 +122,44 @@ export class AppsViewComponent implements OnInit {
     }
   }
 
-  onPaginatorChanged(page: PageEvent) {
+  removeRowSelection() {
     if (this.selectedRow) {
       this.selectedRow.isSelected = false;
       this.selectedRow = null;
       this.allocDataSource.data = [];
     }
   }
 
+  onPaginatorChanged() {
+    this.removeRowSelection();
+  }
+
   isAppDataSourceEmpty() {
     return this.appDataSource.data && this.appDataSource.data.length === 0;
   }
 
   isAllocDataSourceEmpty() {
     return this.allocDataSource.data && this.allocDataSource.data.length === 0;
   }
+
+  onClearSearch() {
+    this.searchText = '';
+    this.removeRowSelection();
+    this.appDataSource.data = this.initialAppData;
+  }
+
+  onSearchAppData() {
```

Comment:
I think a regex based search is something we can plan in next phase.
appid & queueName etc